### PR TITLE
Different backgrounds

### PR DESCRIPTION
**Pull request by 9728627**
I have changed the background when you choose different backgrounds. (100xp)

I realize that this I only changed functions related to this error and other modiftications relataed to other files are not here. It is the correct way to use branch in gitbub. Am I right?

Best 

Xiao Sun

### CODE REVIEW

File: src/edu/ucsb/cs56/projects/games/fish_animation/FishAnimationEnvironment.java [hunk 1]
```diff
@@ -289,8 +297,28 @@ public void paintComponent(Graphics g) {
 			g2.setColor(Color.BLUE);
 			g2.fillRect(0, 0, this.getWidth(), this.getHeight());
 			super.paintComponent(g); // replace current painting
+<<<<<<< Updated upstream
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/fish_animation/FishAnimationEnvironment.java]: The stuff with `<<<<<<<` or `=======` or `>>>>>>>` is left over from merge conflicts.  You'll need to fix that before we can merge.  It probably can't even compile the way it is now.

File: src/edu/ucsb/cs56/projects/games/fish_animation/FishAnimationEnvironment.java [hunk 2]
```diff
@@ -81,8 +81,16 @@
 	private ArrayList<Plankton> plankton = new ArrayList<Plankton>();
 
 	// Images in the game
-	private URL reefURL = getClass().getResource("/resources/CoralReef.jpg");
-	private Image reef = new ImageIcon(reefURL).getImage();
+    //difficulty1
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/fish_animation/FishAnimationEnvironment.java]: Can you use tabs for indentation to match the original file?  It seems to use one tab per level, while you're using four spaces.

File: src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java [hunk 3]
```diff
@@ -612,14 +621,27 @@ public void actionPerformed(ActionEvent event) {
...
+            //change background1
+            diffBG = 1;
+			
+>>>>>>> Stashed changes
 		}
 		if (event.getSource() == Medium) {
 			frame.setVisible(false);
 			FishAnimationEnvironment f = new FishAnimationEnvironment(character_type, 7, false);
+            //change background2
+            diffBG = 2;
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java]: Another indentation note...

File: src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java [hunk 4]
```diff
@@ -33,11 +33,19 @@
...
 	private JButton Character, CMenu, Kwhale, Shark;
-	private JFrame frame, instruct;
+
 
 //	private int type;
 //	private JTextArea text;
 //	private JPanel textpan;
+
+	public static JFrame frame, instruct;
+	public static int diffBG;
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java]: Making this static on `Menu` isn't a great place to put it.  Can you pass this to the constructor of `FishAnimationEnvironment` instead?  Or - even better - can you make `FishAnimationEnvironment` draw the correct background based on the `difficulty` member variable it already has?

File: src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java [hunk 5]
```diff
@@ -33,11 +33,19 @@
...
+	
+	//    private int type;
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java]: Let's clean out this unused code.

File: src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java [hunk 6]
```diff
@@ -33,11 +33,19 @@
 
 	private JButton Play, Instruction, Exit, Resume, Easy, Medium, Hard, Back, Menu, Credit, PlayBGM, PauseBGM, HighScore, NextPage;
 	private JButton Character, CMenu, Kwhale, Shark;
-	private JFrame frame, instruct;
+
 
 //	private int type;
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java]: I know this isn't from your PR, but it's right here...  Can you remove this unused code too?

Comment (9728627) [reply src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java]: you mean I should remove it in my master branch?

File: src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java (hunk 4 above)
Comment (9728627) [reply src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java]: I tried, but failed. If I put them in the constructor and remove them in Menu.java, the program will crash. I think it is better to leave it there.

File: src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java (hunk 6 above)
Comment (xavierholt) [reply src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java]: Nah - let's take it out in this branch (diffBG).

File: src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java (hunk 4 above)
Comment (xavierholt) [reply src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java]: You don't need a `diffBG` variable at all.  Just decide what to draw based on the `difficulty` variable.

As a general rule, treat static variables with _almost_ as much suspicion as you treat global variables.  There's a small chance they're there for a good reason, but most of the time it's just poor design.

Comment (9728627) [reply src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java]: get it. fixed

File: src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java (hunk 6 above)
Comment (9728627) [reply src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java]: it is not from PR. I don knoe why it is here. I just commited it and hope you can check it again.

File: src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java (hunk 5 above)
Comment (9728627) [reply src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java]: cleaned

File: src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java (hunk 6 above)
Comment (9728627) [reply src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java]: i can not push it rn, wait for few minutes

**Comment by 9728627** [comment]
Fixed I think.

File: src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java [hunk 7]
```diff
@@ -33,11 +33,9 @@
 
 	private JButton Play, Instruction, Exit, Resume, Easy, Medium, Hard, Back, Menu, Credit, PlayBGM, PauseBGM, HighScore, NextPage;
 	private JButton Character, CMenu, Kwhale, Shark;
-	private JFrame frame, instruct;
+	public static JFrame frame, instruct;
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java]: What happened here?  They used to be private members, and now they're public statics?  This is almost definitely wrong.  You'll want to revert this.

Comment (9728627) [reply src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java]: it is becasue it was used in my other pull request that I correct the glitchy frames. I have to access the frame from other classes.

Comment (9728627) [reply src/edu/ucsb/cs56/projects/games/fish_animation/Menu.java]: fixed

**Pull request merged**